### PR TITLE
feat: Add support for exclude from summaries.

### PR DESCRIPTION
**Pull request by kinyoklion**
feat: Add `sampleWeight` to flag data model.

### CODE REVIEW

File: packages/shared/common/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "build": "npx tsc",
     "clean": "npx tsc --build --clean",
     "lint": "npx eslint . --ext .ts",
-    "lint:fix": "yarn run lint -- --fix"
+    "lint:fix": "yarn run lint --fix"
```
Comment (keelerm84) [inline packages/shared/common/package.json:27]: Deja vu, we meet again.

File: packages/shared/common/src/internal/events/EventSummarizer.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export default class EventSummarizer {
   private contextKinds: Record<string, Set<string>> = {};
 
   summarizeEvent(event: InputEvent) {
-    if (isFeature(event)) {
+    if (isFeature(event) && !event.excludeFromSummaries) {
```
Comment (keelerm84) [inline packages/shared/common/src/internal/events/EventSummarizer.ts:55]: If `excludeFromSummaries` is optional, that means it defaults to undefined right? And `!undefined == true`? If so, I think this logic might be incorrect.

Comment (kinyoklion) [reply packages/shared/common/src/internal/events/EventSummarizer.ts:55]: `!undefined == true` also `!false == true` and `!true == false`

Comment (keelerm84) [reply packages/shared/common/src/internal/events/EventSummarizer.ts:55]: Ignore me. I'm a fool.

**Pull request merged**